### ISSUE TITLE
ansible winrm Authentication or permission failure

### ISSUE THREAD

**Issue by virain** [hashicorp/packer#9200]
#### Overview of the Issue

Authentication or permission failure. In some cases, you may have been able to authenticate and did not have permissions on the target directory. Consider changing the remote tmp path in ansible.cfg to a path rooted

### Packer version
 `packer 1.5.6`

### Simplified Packer Buildfile

packer.json:
```
{
  "builders": [
    {
      ....
      "communicator": "winrm",
      "winrm_username": "Administrator",
      "winrm_password": "xxxxxx",
      "user_data_file": "winrm_enable_userdata.ps1",
      "winrm_insecure": true,
      "winrm_use_ssl": true
  }
  ],
  "provisioners": [
    {
      "type": "ansible",
      "playbook_file": "main.yml",
      "user": "Administrator",
      "ansible_env_vars": ["no_proxy=\"*\""],
      "extra_arguments": [
        "-e", "ansible_winrm_server_cert_validation=ignore"
      ]
    }
  ]
}
```
main.yml:
```
- name: Windows
  hosts: all
  become: yes
  become_user: Administrator
  become_method: runas
  
  tasks:
    - win_ping:
```

### Operating system and Environment details

`macOS Catalina 10.15.4`
`ansible 2.9.4`

### Log 

```
    alicloud: WinRM connected.
==> alicloud: #< CLIXML
2020/05/09 15:11:24 packer-builder-alicloud-ecs plugin: Connected to machine
2020/05/09 15:11:24 packer-builder-alicloud-ecs plugin: Running the provision hook
2020/05/09 15:11:24 [INFO] (telemetry) Starting provisioner ansible
==> alicloud: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>
==> alicloud: Connected to WinRM!
==> alicloud: Provisioning with Ansible...
    alicloud: Setting up proxy adapter for Ansible....
2020/05/09 15:11:24 packer-provisioner-ansible plugin: Creating inventory file for Ansible run...
2020/05/09 15:11:24 packer-provisioner-ansible plugin: SSH proxy: serving on 127.0.0.1:60485
==> alicloud: Executing Ansible: ansible-playbook -e packer_build_name=alicloud -e packer_builder_type=alicloud-ecs -e ansible_ssh_private_key_file=/var/folders/h3/n45fh_c16v7956m1v027nnz80000gn/T/ansible-key779486652 -i /var/folders/h3/n45fh_c16v7956m1v027nnz80000gn/T/packer-provisioner-ansible981475819 /Users/osmarst/win_packer/main.yml -e ansible_winrm_server_cert_validation=ignore
    alicloud:
    alicloud: PLAY [Windows] *****************************************************************
    alicloud:
    alicloud: TASK [Gathering Facts] *********************************************************
2020/05/09 15:11:25 packer-provisioner-ansible plugin: SSH proxy: accepted connection
2020/05/09 15:11:25 packer-provisioner-ansible plugin: authentication attempt from 127.0.0.1:60489 to 127.0.0.1:60485 as Administrator using none
2020/05/09 15:11:25 packer-provisioner-ansible plugin: authentication attempt from 127.0.0.1:60489 to 127.0.0.1:60485 as Administrator using publickey
2020/05/09 15:11:25 packer-provisioner-ansible plugin: new env request: LC_TERMINAL_VERSION=3.3.9
2020/05/09 15:11:25 packer-provisioner-ansible plugin: new env request: LANG=zh_CN.UTF-8
2020/05/09 15:11:25 packer-provisioner-ansible plugin: new env request: LC_TERMINAL=iTerm2
2020/05/09 15:11:25 packer-provisioner-ansible plugin: new exec request: /bin/sh -c 'echo ~Administrator && sleep 0'
2020/05/09 15:11:25 packer-builder-alicloud-ecs plugin: [INFO] starting remote command: /bin/sh -c 'echo ~Administrator && sleep 0'
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] command '/bin/sh -c 'echo ~Administrator && sleep 0'' exited with code: 1
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] RPC endpoint: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] 0 bytes written for 'stdout'
2020/05/09 15:11:26 [INFO] 44 bytes written for 'stderr'
2020/05/09 15:11:26 [INFO] RPC client: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] RPC endpoint: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] 0 bytes written for 'stdin'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 0 bytes written for 'stdout'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 44 bytes written for 'stderr'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] RPC client: Communicator ended with: 1
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 0 bytes written for 'stdin'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new env request: LC_TERMINAL_VERSION=3.3.9
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new env request: LANG=zh_CN.UTF-8
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new env request: LC_TERMINAL=iTerm2
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new exec request: /bin/sh -c 'echo "`pwd`" && sleep 0'
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] starting remote command: /bin/sh -c 'echo "`pwd`" && sleep 0'
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] command '/bin/sh -c 'echo "`pwd`" && sleep 0'' exited with code: 1
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] RPC endpoint: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] 0 bytes written for 'stdout'
2020/05/09 15:11:26 [INFO] 44 bytes written for 'stderr'
2020/05/09 15:11:26 [INFO] RPC client: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] RPC endpoint: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] 0 bytes written for 'stdin'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 0 bytes written for 'stdout'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 44 bytes written for 'stderr'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] RPC client: Communicator ended with: 1
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 0 bytes written for 'stdin'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new env request: LC_TERMINAL_VERSION=3.3.9
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new env request: LANG=zh_CN.UTF-8
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new env request: LC_TERMINAL=iTerm2
2020/05/09 15:11:26 packer-provisioner-ansible plugin: new exec request: /bin/sh -c '( umask 77 && mkdir -p "` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `" && echo ansible-tmp-1589008285.801184-269357992766933="` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `" ) && sleep 0'
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] starting remote command: /bin/sh -c '( umask 77 && mkdir -p "` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `" && echo ansible-tmp-1589008285.801184-269357992766933="` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `" ) && sleep 0'
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] command '/bin/sh -c '( umask 77 && mkdir -p "` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `" && echo ansible-tmp-1589008285.801184-269357992766933="` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `" ) && sleep 0'' exited with code: 1
2020/05/09 15:11:26 packer-builder-alicloud-ecs plugin: [INFO] RPC endpoint: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] 0 bytes written for 'stdout'
2020/05/09 15:11:26 [INFO] 44 bytes written for 'stderr'
2020/05/09 15:11:26 [INFO] RPC client: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] RPC endpoint: Communicator ended with: 1
2020/05/09 15:11:26 [INFO] 0 bytes written for 'stdin'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 0 bytes written for 'stdout'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 44 bytes written for 'stderr'
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] RPC client: Communicator ended with: 1
2020/05/09 15:11:26 packer-provisioner-ansible plugin: [INFO] 0 bytes written for 'stdin'
    alicloud: fatal: [default]: UNREACHABLE! => {"changed": false, "msg": "Authentication or permission failure. In some cases, you may have been able to authenticate and did not have permissions on the target directory. Consider changing the remote tmp path in ansible.cfg to a path rooted in \"/tmp\". Failed command was: ( umask 77 && mkdir -p \"` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `\" && echo ansible-tmp-1589008285.801184-269357992766933=\"` echo ~/.ansible/tmp/ansible-tmp-1589008285.801184-269357992766933 `\" ), exited with result 1", "unreachable": true}
    alicloud:
    alicloud: PLAY RECAP *********************************************************************
    alicloud: default                    : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
    alicloud:
2020/05/09 15:11:26 packer-provisioner-ansible plugin: shutting down the SSH proxy
2020/05/09 15:11:26 [INFO] (telemetry) ending ansible
==> alicloud: Error executing Ansible: Non-zero exit status: exit status 4
```

**Comment by nywilken** [comment]
Hi @virain thanks for reaching out. I don't have enough experience with the ansible provisioner to call out exactly what the issue is here. But I suspect you may be running into a previous issue that was fixed by the introduction of the `use_proxy` configuration option. If you try adding `"use_proxy": false` to your Ansbile configuration does it resolve the issue? 

More information on the how the provisioner behaves with setting `use_proxy` to false can be found at https://www.packer.io/docs/provisioners/ansible/#use_proxy

**Comment by MilanNXT** [comment]
I believe this is relate to the same issue that [#7224](https://github.com/hashicorp/packer/issues/7224)
In private setup you using proxy configuration to allow az cli and packer to reach public endpoints. In the same time winrm is also sent through proxy which in many case is disabled by company. We need to implement winrm_no_proxy option. It was implemented [#5725](https://github.com/hashicorp/packer/issues/5725) but also immediately reverted back.

**Comment by virain** [comment]
👌
